File: BE/src/model/user.model.ts
  import mongoose, { Document, Schema, model } from "mongoose";
  import { UserRole, UserStatus } from "../enum/user.enum"
  export interface IUser {
    name: string;
    email: string;
    password: string;
    role: "admin" | "teacher" | "student";
    avatar?: string;
    status: "active" | "locked";
    description?: string;
    createdAt: Date;
    updatedAt: Date;
    lastLogin?: Date;
  }

  export interface IUserDocument extends IUser, Document {}

  const userSchema = new Schema<IUserDocument>(
    {
      name: { type: String, required: true },
      email: { type: String, required: true, unique: true },
      password: { type: String },
      role: {
        type: String,
        enum: Object.values(UserRole),
        default: UserRole.STUDENT,
      },
      avatar: String,
    status: {
        type: String,
        enum: Object.values(UserStatus),
        default: UserStatus.ACTIVE,
      },
      description: { type: String },
      lastLogin: { type: Date, default: null },
    },
    { timestamps: true }
  );

  // userSchema.set("toJSON", {
  //   transform: function (doc, ret: any) {
  //     ret.id = ret._id;
  //     delete ret._id;
  //     delete ret.__v;
  //     delete ret.password;
  //     return ret;
  //   },
  // });

  export const User = mongoose.model<IUserDocument>("User", userSchema);

File: mirai_mobile/lib/models/user_model.dart
class UserModel {
  final String id;
  final String email;
  final String name;
  final String? avatar;
  final String role;
  final String? status;

  UserModel({
    required this.id,
    required this.email,
    required this.name,
    this.avatar,
    required this.role,
    this.status,
  });

  factory UserModel.fromJson(Map<String, dynamic> json) {
    return UserModel(
      id: json['_id'] as String? ?? json['id'] as String? ?? '',
      email: json['email'] as String? ?? '',
      name: json['name'] as String? ?? '',
      avatar: json['avatar'] as String?,
      role: json['role'] as String? ?? 'user',
      status: json['status'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'email': email,
      'name': name,
      'avatar': avatar,
      'role': role,
      'status': status,
    };
  }
}
